### ISSUE TITLE
Read preference tags are not picked up from a mongo uri

### ISSUE THREAD

**Issue by tmichel** [Automattic/mongoose#4311]
When connecting with a mongo uri the specification allows for specifying `readPreferenceTags` as a query parameter. According to the docs the `readPreferenceTags` parameter is supported in the uri but it is not picked up.

I tried to chase how the configuration is read and passed around and it seems that it is correctly read from the uri (using the `muri` package) and set as `read_preference_tags` in the options object. This field is then never picked up by any other component that could use the read preference tags.

There is an easy workaround though, you can pass in the tags directly:

```javascript
mongoose = require('mongoose')

mongoose.connect("mongodb://localhost:27017,localhost:27018/produts", {
  db: {
    readPreference: {
      preference: 'secondaryPreferred',
      tags: [{use: "reporting"}]
    }
  }
})
```

Also be aware that there is a bug in the `mongodb-core` package that entirely prevents using tags. [There is a pull request][1] with the fix, and hopefully it will be merged quickly. 

[1]: https://github.com/christkv/mongodb-core/pull/107

**Comment by vkarpov15** [comment]
Yeah unfortunately mongoose handles URI parsing in it's own special way, so we'll probably need to do some work in mongoose to make this behave once the mongodb-core PR is merged.

**Comment by tmichel** [comment]
Apparently the tagging issue is fixed on mongodb-core 2.x. I don't know if it is feasible to update the dependency to pick up the newer version of mongodb-core. Actually the mongodb should be updated since that is the direct dependency of mongoose.

**Comment by vkarpov15** [comment]
Haven't investigated that yet, will check when I look at this issue.

**Comment by vkarpov15** [comment]
Fixed by #4363
